### DISCUSSION TITLE
Ballast Flora needs changes

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#4877] [Ideas and suggestions]
When the ballast flora first released, I was excited to encounter this new enemy, but as I progressed through the campaign, I became increasingly frustrated every time my ballast was infected. Infection didn't really associate with "challenge", but rather "tedium". 

Clearing ballast Flora had become a painful, resource consuming routine which left little room for enjoyment or improvisation. As soon as the ideal procedure was set up, you had to waste a good 5 minutes distributing tools, setting up the procedure, clearing the flora and applying large amounts of bandages to the crew. That, or chugging several grenades down the ballast, which for some reason are more effective than setting the whole thing on fire or blowing up C4 near your pumps.

_In short,_ here are the issues with Ballast Flora: It's predictable, it's non-threatening, and it's a resource-check. It always spawns in the same place, the power drain does very little to threaten your ship until the next outpost, and each infection means siphoning a set amount of resources (fuel, bandages...). 

So here are 3 proposed changes changes for Flora:

-**Flora needs to be a more active hazard.** Currently flora is nothing more than an annoyance should you choose to ignore it. This is because flora infects the ballast, a low-priority room, and as it spreads, only drains power extra power which has little impact on the sub. It doesn't actually present a risk until you actively deal with it, which can wait the next outpost. Flora needs to either actively damage JBs, passively emit gas, or prevent machinery repairs. Flora actively preventing ship maintenance combined with external events will make it a truly dangerous element if ignored

-**Flora should infect any kind of pump.** Flora being restricted to the ballast means it's predictable and clearing is only a matter of procedure. Most subs only have 2 or 3 ballasts, often clustered which considerably limits where flora can spawn and infect. However, spawning flora even on bilge pumps means more infection patterns, potentially disabling different rooms and more unpredictable behavior, and a different approach for clearing it as you wouldn't be able to just throw grenades or set the room on fire should it infect the oxygen storage or high traffic areas.

-**Flora should create new, weaker "root" nodes as it expands.** Right now the only viable way to deal with flora is rushing the root at the pump. There is currently no reason to clear the branches either; you only focus on the root. Duplicating "roots" means there wouldn't be a singular weak point that kills all the flora. It needs more, less resilient roots spread across the rooms.

Overall Flora needs to be less of a tedious cleaning procedure and more of an active, unpredictable hazard.

**Comment by LupusTheCanine** [comment]
Flora infestation check should only happen on flow in, not out for  any pump, bilge pumps would typically be protected by one way valve to prevent backflow in case of malfunction anyway.

**Reply by Xeavas** [reply]
That's already a thing afaik, ballast flora can only trigger if the given pump is pumping in water at the time of the "trigger".

**Reply by ubertpendragon** [reply]
All pumps should be infect-able for gameplay purposes, doesn't matter if they flow in or out

**Reply by CurePindal** [reply]
> 
> 
> All pumps should be infect-able for gameplay purposes, doesn't matter if they flow in or out

Ballast flora is a fantastic concept to expand on. Im on board with this. And enhanced spreading of flora adds to the game and nobody would get hurt. win-win in my book.

**Comment by Singularia** [comment]
I agree that ballast flora should be more aggressive and have more effects on installations (like slowing crafting speed, pumping speed, engine output, interrupting maintenance attempts etc). I also agree that it should create new roots, to make it less vulnerable to skilled players (because if you can identify correct source pump then it will be an easy* kill). Also i think delay between successful infection and actual flora spawn should be more randomized, because currently you can easily predict when exactly it will spawn.

And i fully disagree about ability to infect all pumps. Bilge pumps should be a separate system that could be infected ONLY if flora got them from inside. If flora would be able to infect all pumps then submakers will ether put them in isolated parts like ballast ones or will switch to ducts Humpback style. Basically workaround will be found is what i am saying.

Maybe there should be more entry points though, not just one ballast pump but 2 or more

**Comment by MaaiG** [comment]
- I don’t like when the flora coming through closed doors. If the bay is sealed and the door doesn’t open, why is the flora spreading? No fit!
- Ballast flora is pretty basic. Turning off the ballast compartment and throwing a `Incendium Grenade`. When all flora will be burns, just turn on the ballast compartment.
- Roots must propagate and create new root nodes in random locations. These roots are just as strong and can be major.
- Floral flowers can be dangerous. Gas may be released or live organisms may be attacked.
- Aircrew contamination.
- Interference and equipment malfunction.
- The flora may be secretive and its destruction may not be obvious.
- Flora can wear parasites, diseases down to husk and worse...
- Flora can adapt. Some species of flora are not exposed to fire or other. Look for a `Chemist` and chemical components and formulas to control it. Do research if you’re unlucky.
- If you neglect the biohazard... you may have to abandon your ship! Or die!
- You’ll have to sink your beloved ship and you can’t have anything to remember you by. But you will remembered forever - The Ballast Flora!
- Etc.

[Sleep swetie or she’ll come for you!](https://www.youtube.com/watch?v=5ftmr17M-a4&t=80s)
